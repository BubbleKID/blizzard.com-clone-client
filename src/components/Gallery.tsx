import Card from './Card';
import diablo2Cover from '../images/fenway-game-card.jpeg';
import overWatch2Cover from'../images/game-card_overwatch2_enUS.jpeg';
import overWatchCover from'../images/game-card_overwatch_enUS.jpeg';
import wowCover from '../images/game-card_wow_enUS.jpeg';

export default function Gallery() {
    const cards = [{ name: "Diablo2", background: diablo2Cover},
      { name: "Overwatch2", background:overWatch2Cover },  
      { name: "Overwatch", background:overWatchCover },
      { name: "World of Warcraft", background:wowCover }
    ];
     
    return (
      <section>
        <h1>Games</h1>
        {cards.map(card => (
            <Card key={card.name} background={card.background} name={card.name}></Card>
          ))
        }
      </section>
    );
}