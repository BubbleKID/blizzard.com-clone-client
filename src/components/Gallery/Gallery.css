.gallery-container {
    background-position: top center;
    background-repeat: no-repeat;
}

.gallery {
    display: flex;
    margin: 0 auto;
    justify-content: center;
    max-width: 1170px;
}

.headdingContainer {
    display: flex;
    justify-content: space-between;
    align-items: baseline;
    margin: 0 12px;
}

.headdingContainer a {
    text-decoration: none;
}