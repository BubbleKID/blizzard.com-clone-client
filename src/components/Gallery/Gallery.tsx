import Card from '../Card/Card';
import './Gallery.css';
import diablo2Cover from '../../images/fenway-game-card.jpeg';
import overWatch2Cover from'../../images/game-card_overwatch2_enUS.jpeg';
import overWatchCover from'../../images/game-card_overwatch_enUS.jpeg';
import wowCover from '../../images/game-card_wow_enUS.jpeg';
import hsCover from '../../images/game-card_hearthstone_enUS.jpeg';
import heroesCover from '../../images/game-card_heroes_enUS.jpeg';
import war3Cover from '../../images/game-card-wc3-en-us.jpeg';
import diablo4 from '../../images/game-card_diablo4_enUS.jpeg';
import diabloimmortalCover from '../../images/game-card_diabloimmortal_enUS.jpeg';
import diablo3Cover from '../../images/game-card_diablo3_enUS.jpeg';
import sc2Cover from '../../images/game-card_starcraft2_enUS.jpeg';
import scCover from '../../images/game-card_starcraft_enUS.jpeg';
import arcadeCover from '../../images/rtro-game-card.jpeg';
import background from '../../images/background-generic-landscape-short.jpg'

export default function Gallery() {
    const cards = [
      { name: "Diablo2", background: diablo2Cover},
      { name: "Overwatch2", background: overWatch2Cover, url: '#' },  
      { name: "Overwatch", background: overWatchCover, url: '#' },
      { name: "World of Warcraft", background: wowCover, url: '#' },
      { name: "HearthStone", background: hsCover, url: '#' },
      { name: "Heroes of The Storm", background: heroesCover, url: '#' },
      { name: "Warcraft3", background: war3Cover, url: '#' },
      { name: "Diablo4", background: diablo4, url: '#' },
      { name: "Diablo Immortal", background: diabloimmortalCover, url: '#' },
      { name: "Diablo3", background: diablo3Cover, url: '#' },
      { name: "Starcraft2", background: sc2Cover, url: '#' },
      { name: "Starcraft", background: scCover, url: '#' },
      { name: "Blizzard Arcade", background: arcadeCover, url: '#' }
    ];
     
    return (
      <div className="gallery-container" style={{backgroundImage: `url(${background})`}}>
        <div className="gallery-wrapper">
          <div className='headdingContainer'>
            <h2 className="heading">Games</h2>
            <a><span>Shop All Games</span></a>
          </div>
          <div className='gallery'>
            {cards.map(card => (
                <Card key={card.name} background={card.background} name={card.name}></Card>
              ))
            }
          </div>
        </div>
      </div>
    );
}