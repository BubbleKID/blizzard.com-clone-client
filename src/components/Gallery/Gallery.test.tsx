import React from 'react';
import TestRenderer  from 'react-test-renderer';
import Gallery from './Gallery';

test('Gallery should be rendered', () => {
    const component = TestRenderer.create(<Gallery/>,);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});