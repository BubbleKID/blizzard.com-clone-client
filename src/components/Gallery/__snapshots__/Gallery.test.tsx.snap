// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should be rendered 1`] = `
<div
  className="gallery-container"
  style={
    Object {
      "backgroundImage": "url()",
    }
  }
>
  <div
    className="gallery-wrapper"
  >
    <div
      className="headdingContainer"
    >
      <h2
        className="heading"
      >
        Games
      </h2>
      <a>
        <span>
          Shop All Games
        </span>
      </a>
    </div>
    <div
      className="gallery"
    >
      <a
        className="card"
        href="#"
      >
        <img
          alt="Diablo2"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Overwatch2"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Overwatch"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="World of Warcraft"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="HearthStone"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Heroes of The Storm"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Warcraft3"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Diablo4"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Diablo Immortal"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Diablo3"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Starcraft2"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Starcraft"
          src=""
        />
      </a>
      <a
        className="card"
        href="#"
      >
        <img
          alt="Blizzard Arcade"
          src=""
        />
      </a>
    </div>
  </div>
</div>
`;
